Directory setup

hiromon0125 opened the pull request:
#Feature
* Directory for raspberry pi
* Directory for arduino
* Directory for catbot-cli
* ReadMe file for each directories mentioned above
* Master readMe file with
  - [x] short descriptions
  - [x] setup instructions
  - [x] command instructions

hiromon0125 committed:
feat: add readme files

hiromon0125 committed:
fix: get rid of version number

mdrc-catbot committed:
feat: initialize arduino sketch

hiromon0125 committed:
fix: move sketch to correct directory

hiromon0125 committed:
feat: executable catbot cli

hiromon0125 committed:
feat: add readme files and mv requirements to root

hiromon0125 committed:
fix: readme

hiromon0125 committed:
feat: mdrc logo

mdrc-catbot committed:
feat: package-lock

thatnoobles committed:
stubbed out motor & pneumatics classes

thatnoobles committed:
stubbed out some of receiver

thatnoobles committed:
created basic adruino header files

cvr8924 reviewed (dismissed):
Looks good to me. I want to have a talk abt the CLI stuff but you should be good to push to Dev

thatnoobles reviewed (approved):
Looks good - I don't see anything missing

thatnoobles committed:
stubbed out states for possible eventhandler + log

hiromon0125 committed:
feat: parse board infos

thatnoobles committed:
added basic docstrings for (most) things

hiromon0125 committed:
remove: cli temporary for file structure merge

thatnoobles committed:
Update src/raspi/io/pneumatics/valve.py

set_value should pass 'self' as a parameter

Co-authored-by: Hiroto Takeuchi <hiroto125takeuchi@gmail.com>

thatnoobles committed:
Update src/arduino/comms/packager.h

Little TODO message for adding docstrings to the C++ code if needed eventually

Co-authored-by: Hiroto Takeuchi <hiroto125takeuchi@gmail.com>

thatnoobles committed:
Merge pull request #5 from RIT-MDRC/stubbing

Stubbing

cvr8924 commented on src/arduino/README.md:
```diff
@@ -0,0 +1,6 @@
+## Description
+This folder contains the code for the Arduino Nano. The code is written in C++ and is compiled using the Arduino IDE.
+
+## Short system description
```
This documentation is no longer accurate and should be amended

cvr8924 commented on src/arduino/comms/packager.h:
```diff
@@ -0,0 +1,20 @@
+#include <potentiometer.h>
+#include <packet.h>
+// TODO: docstring
+class Packager
```
This is likely all depreciated by new architecture

cvr8924 replied:
Although some version of this should maybe exist for Future-proofing IMU stuff

thatnoobles replied:
Removed references to potentiometers and pressure in the C++ Packager class. Now acts as more of a general data-transfer tool - once implemented it can be used for the pressure handshake and IMU stuff if we want to.

cvr8924 commented on src/arduino/comms/packet.h:
```diff
@@ -0,0 +1,13 @@
+class Packet
```
We do still need some I2C Packet protocol for futureproofing so while packager needs overhaul prob not this class

thatnoobles replied:
Generalized the naming here a bit. Not much else to do here

cvr8924 commented on src/arduino/components/potentiometer.h:
```diff
@@ -0,0 +1,14 @@
+class Potentiometer
```
Again no longer handled on Arduino, a rasp pi version of this is needed.

Gonna cc @thatnoobles  on these btw cause I don't know what came in from his PR

thatnoobles replied:
Removed references to potentiometers in the C++ code since the Arduino no longer knows about any of that. I've updated the RPi class to reflect that it is going to be receiving potentiometer data from the ADC now.

cvr8924 commented on src/arduino/components/pressure.h:
```diff
@@ -0,0 +1,9 @@
+class Pressure
```
Should there be check pressure in range things for this class? is that a function we need here? or somewhere else

cvr8924 commented on src/raspi/README.md:
```diff
@@ -0,0 +1,6 @@
+## Description
+This folder contains the code for the Raspberry Pi. The code is written in Python and is compiled using the Python interpreter.
+
+## Short system description
```
Also depreciated documentation

cvr8924 commented on src/raspi/io/comms/receiver.py:
```diff
@@ -0,0 +1,6 @@
+
+current_data : list[int]
+potentiometers : list[Potentiometer]
+pressure : Pressure
```
The rasp pi doesn't know the pressures anymore

cvr8924 commented on src/raspi/io/pneumatics/pressure.py:
```diff
@@ -0,0 +1,26 @@
+class Pressure():
```
Rasp pi doesn't know this anymore

cvr8924 commented on src/raspi/io/pneumatics/pressure.py:
```diff
@@ -0,0 +1,26 @@
+        """
+        return False
+    
+    def get_average_pressure(self) -> float:
```
Especially don't know this

thatnoobles replied:
Removed the Pressure class from the RPi code. Since it now only sees a TRUE/FALSE for whether or not the pressure is good (and this dictates whether other behaviors can run), this could be a global variable somewhere that a main loop can write to every "cycle".

cvr8924 reviewed (changes requested):
Hey so a lot of this has been depreciated by new architecture. I'd wanna see some changes to those things before

thatnoobles committed:
re: cam's comments on pull request #4

thatnoobles committed:
removed reference to pressure in packager.h

thatnoobles committed:
re-added AddByte(s) in packet.h

thatnoobles committed:
updated rpi pot class to reflect new architecture

hiromon0125 committed:
fix: readme to not include non-implemented system

hiromon0125 commented:
@cvr8924 I got rid of the system description so that we aren't predefining something that is not yet implemented.

cvr8924 reviewed (approved):
Looks good, thanks for the changes
